### PR TITLE
2.x => 3.x merge and fix tests

### PR DESCRIPTION
**Pull request by LordSimal**

### CODE REVIEW

**Comment by LordSimal** [comment]
@ADmad I'd need your help with this named argument finder change. How do we handle this in the OrmResolver?

File: src/Authenticator/CookieAuthenticator.php [hunk 1]
```diff
@@ -28,6 +28,7 @@
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use RuntimeException;
+use function Cake\Core\deprecationWarning;
```
Comment (ADmad) [inline src/Authenticator/CookieAuthenticator.php]: All deprecations can be simply dropped for 3.x.

**Comment by ADmad** [comment]
> @ADmad I'd need your help with this named argument finder change. How do we handle this in the OrmResolver?

Just changing the [following line](https://github.com/cakephp/authentication/blob/3.x/src/Identifier/Resolver/OrmResolver.php#L66) from `$query->find($finder, $options);` to `$query->find($finder, ...$options);` should suffice.

**Comment by ADmad** [comment]
I took care of the finder changes.

File: src/Authenticator/CookieAuthenticator.php (hunk 1 above)
Comment (LordSimal) [reply src/Authenticator/CookieAuthenticator.php]: Right, forgot that 🙈

**Pull request merged**